docs/blog/2023-03-31-zephyr-3-2.md

```diff
@@ -0,0 +1,267 @@
+
+```
+CONFIG_BT_GATT_ENFORCE_SUBSCRIPTION=n
+```
+
+## Keymap Changes
+
+Due to conflicts with new devicetree node labels added for Zephyr's [reset system](https://example.com/TODO), the `&reset` behavior has been renamed to `&sys_reset`.
+
+All of the in-tree keymaps have been fixed, but you may encounter build failures about duplicate names, requiring you rename the behavior reference in your keymap. Use the [Keymap Upgrader](docs/codes/keymap-upgrader) and this will get fixed for you automatically.
+
+## Board/Shield Changes
+
+The following changes have [already been completed](https://github.com/zmkfirmware/zmk/pull/1143/commits) for all boards/shields in ZMK `main` branch. For existing or new PRs, or out of tree boards, the following changes are necessary to properly work with the latest changes.
+
+### Move to `pinctrl` driver
+
+Before this change, setting up the details of pins to use them for peripherals like SPI, I2C, etc. was a mix of platform specific driver code. Zephyr has moved to the newer `pinctrl` system to unify the handling of pin configuration, with additional flexibility for things like low power modes for those pins, etc.
+
+#### Board specific shield overlays
+
+The main area this affects existing shields is those with board specific overrides, e.g. `<shield>/boards/seeeduino_xiao_ble.overlay`, that sets up additional components on custom busses, e.g. addressable RGB LEDs leveraging the SPI MOSI pin.
+
+#### nRF52 Pin Assignments
+
+Previously in ZMK, we relied on per-driver devicetree source properties to set the alternate pin functions for things like SPI or I2C. For example, here is the I2C bus setup as it was previously on the `nice_nano` board:
+
+```
+&i2c0 {
+    compatible = "nordic,nrf-twi";
+    sda-pin = <17>;
+    scl-pin = <20>;
+};
+```
+
+With the move to the `pinctrl` system, this setup now look like:
+
+```
+ &i2c0 {
+    compatible = "nordic,nrf-twi";
+    pinctrl-0 = <&i2c0_default>;
+    pinctrl-1 = <&i2c0_sleep>;
+    pinctrl-names = "default", "sleep";
+ };
+```
+
+which references the `pinctrl` configuration:
+
+```
+&pinctrl {
+       i2c0_default: i2c0_default {
+               group1 {
+                       psels = <NRF_PSEL(TWIM_SDA, 0, 17)>,
+                               <NRF_PSEL(TWIM_SCL, 0, 20)>;
+               };
+       };
+
+       i2c0_sleep: i2c0_sleep {
+               group1 {
+                       psels = <NRF_PSEL(TWIM_SDA, 0, 17)>,
+                               <NRF_PSEL(TWIM_SCL, 0, 20)>;
+                       low-power-enable;
+               };
+       };
+};
+```
+
+Although slightly more _verbose_ this allows pin configuration infrastructure to be re-used, specify other modes, like sleep, etc. in a standard way across architectures.
+
+#### Out of Tree Boards/Shields
```

Comment:
Need to mention adding `CONFIG_PINCTRL=y` to your _defconfig file.

Reply:
Good catch. Added.